src/OnChain.ts

```diff
@@ -251,64 +256,75 @@ export const OnChainMixin = (superclass) => class extends superclass {
 
   async getOnchainReceipt({confirmed}: {confirmed: boolean}) {
     
-    // optimization to remove the need to fetch lnd when no address
-    // mainly useful for testing purpose
-    // we could only generate an onchain_address the first time the client request it
-    // as opposed to the first time the client log in
-    if (!this.user.onchain_addresses.length) {
-      return []
-    }
-
-    const lnd_incoming_txs = await getOnChainTransactions({ lnd, incoming: true })
-    
-    // for unconfirmed tx: 
-    // { block_id: undefined,
-    //   confirmation_count: undefined,
-    //   confirmation_height: undefined,
-    //   created_at: '2021-03-09T12:55:09.000Z',
-    //   description: undefined,
-    //   fee: undefined,
-    //   id: '60dfde7a0c5209c1a8438a5c47bb5e56249eae6d0894d140996ec0dcbbbb5f83',
-    //   is_confirmed: false,
-    //   is_outgoing: false,
-    //   output_addresses: [Array],
-    //   tokens: 100000000,
-    //   transaction: '02000000000...' }
+    const pubkeys: string[] = this.user.onchain_pubkey
+    let user_matched_txs: GetChainTransactionsResult["transactions"] = []
```

Comment:
Not sure if it's a good practice to use typings from other packages in this manner. It's hard to find the type without digging into the `Lightning` package's code. While it probably isn't a problem in an editor like VS Code (with it's Ctrl+click functionality), I think it makes the dev experience slightly worse if you're using a lighter editor or reviewing on web.

Reply:
do you have a recommendation here?
expect duplicating the type to our repo I'm not sure what other there would be. 

on the code editor side, I think that when working with Typescript, it doesn't make a lot of sense to use an editor that doesn't provide type hints.

Reply:
> do you have a recommendation here?
> expect duplicating the type to our repo I'm not sure what other options there would be.

Nope, not apart from duplicating it.

> on the code editor side, I think that when working with Typescript, it doesn't make a lot of sense to use an editor that doesn't provide type hints.

Yeah kinda makes sense. It will still be hard for someone who's just looking through the code on github as opposed to cloning and viewing. We'll have a lot more people in the former category than the latter. Something to consider.

Reply:
Maybe a comment that has the structure of the type (an example) makes sense. This is something we've already been using and it definitely helps imo

Reply:
I think in general we should start refactoring to not rely on the third party dependencies on so many places in the code.

Have 1 file that talks to the dependency and acts as a wrapper exposing our own types internally. That way if things change upstream you only have to touch the wrapper and not the rest of the code.

But this obviously requires quite a bit of cleanup and is best done step-by-step.